core/modules/util/EventThread.cc

```diff
@@ -42,6 +42,17 @@ namespace lsst {
 namespace qserv {
 namespace util {
 
+
+void CommandQueue::queCmd(std::vector<Command::Ptr> const& cmds) {
+    {
+        std::lock_guard<std::mutex> lock(_mx);
+        for (auto const& cmd:cmds) {
+            _qu.push_back(cmd);
+        }
+    }
+    notify(cmds.size() > 1); // notify all if more than 1 command, otherwise notify 1.
```

Comment:
I'm still convinced that this should be handled via a semaphore that gives you precise control of how many threads will be usefully engaged. My presumption is that more often than note the number of tasks will be >1 but practically always less that the number of waiting threads.